Convert from Webpack to Vite

ChaosExAnima opened the pull request:
This is a PR based on #24981 that will merge into #34454. It also contains #34469 and #34485.

ChaosExAnima committed:
changes entrypoints and tweaks manifest augmenter

ChaosExAnima committed:
add asset integrity key to file

ChaosExAnima committed:
remove references to webpack

ChaosExAnima committed:
move postcss to vite config, and get rid of more WP dependencies

ChaosExAnima committed:
Migrate from Jest to Vitest

ChaosExAnima committed:
add all entrypoints and inserts module polyfill, plus loads Vite in dev mode

renchap commented on package.json:
```diff
@@ -186,23 +152,23 @@
     "stylelint": "^16.11.0",
     "stylelint-config-prettier-scss": "^1.0.0",
     "stylelint-config-standard-scss": "^14.0.0",
+    "ts-jest": "^29.3.2",
```
I am not sure why, but you added back `ts-jest` and `@types/jest`, I dont think its expected as this PR contains the switch to Vite?

ChaosExAnima committed:
Webpack to Vite: Utilize Ruby Vite (#34469)

Co-authored-by: Renaud Chaput <renchap@gmail.com>

ChaosExAnima committed:
add preval again and fix import issues with build

vmstan commented:
![CleanShot 2025-05-06 at 16 47 37@2x](https://github.com/user-attachments/assets/97ef9dff-b044-4f09-940b-cde2af490763)

Live on vmst.io, so far so good.

vmstan commented:
I just took the patches off an hour ago since they were barking about merge errors when I tried to redeploy.

renchap commented on app/views/layouts/application.html.haml:
```diff
@@ -26,11 +26,12 @@
-    = stylesheet_pack_tag 'inert', media: 'all', crossorigin: 'anonymous', id: 'inert-style'
+    = vite_stylesheet_tag 'styles/inert.scss', media: 'all', id: 'inert-style'
 
-    = javascript_pack_tag 'common', crossorigin: 'anonymous'
```
We should ensure that `common` did not execute any JS side effect, or load any CSS (I dont think I checked this)

ChaosExAnima replied:
`common` as in `common.js`? the only thing that file does is start the Rails js file.

renchap replied:
I think that `rails-ujs` runs some side effects, so we need to ensure that it is imported on every page, but as think that its the case because we call `start()` in `application.ts`, and this file is included everywhere `common` was previously called?

… and after checking, the `common` chunk was special in the webpack configuration, and was the default chunk. So as long as we still call `start()` in every "main" entrypoint (the ones that are loaded from Rails layouts so that we have at least one of those on every page), we should be good

ChaosExAnima replied:
so right now it's imported by `application.ts`, `embed.tsx`, `public.tsx`, and `share.tsx`. i'm not sure what you mean by "main". here are the ones that don't import it that i can see:

* `admin.tsx`
* `error.ts`
* `remote_application_helper.ts`
* `sign_up.tsx`
* `two_factor_authentication.ts`

ChaosExAnima replied:
fixed with c7b039ce81a00fd982ef8057fa8f8e9e4c9abeb4. i imported common everywhere, but perhaps it'd be fine to rely on just `application.html.haml`?

renchap commented on app/views/shared/_web_app.html.haml:
```diff
@@ -1,15 +1,16 @@
-    = preload_pack_asset 'features/compose.js'
-    = preload_pack_asset 'features/home_timeline.js'
-    = preload_pack_asset 'features/notifications.js'
+    -#
```
Let's remove this

renchap commented on babel.config.js:
```diff
@@ -1,80 +0,0 @@
-    corejs: { version: "3.30" },
-    debug: false,
-    include: [
-      'transform-numeric-separator',
```
I did not check, but iirc there was a reason for those to be here so old browsers do not choke on those syntaxes. Maybe its no longer needed with Vite, but I remember that we had to add those explicitely to have babel transform those features.

ChaosExAnima replied:
i checked the output, and it looks like Vite rewrites the numbers by default, so this:
```ts
const TEST = 100_000_000;
const TEST2 = 1_0;
```
becomes:
```js
const T=1e8;
const T2=10;
```

renchap replied:
Ok, can you check for the other ones as well? I remember that we had issues because the nullish coalescing operator (`??`) was not properly transpiled in some browsers.

But if this is no longer the case, then maybe it means we need to update our `browserlistrc` to include those browsers rather than manually picking things to transpile?

ChaosExAnima replied:
that seems untouched. if we want to support this, we should update our `browserlistrc` as all the things mentioned in this config have been supported for a while:

* https://caniuse.com/mdn-javascript_operators_nullish_coalescing
* https://caniuse.com/mdn-javascript_operators_optional_chaining
* https://caniuse.com/mdn-javascript_classes_private_class_methods

unless we we want to support IE11 i think it's fine to drop these.

renchap replied:
Ok I looked back at the git history, and those were here because webpack or the minifier did not understood those syntaxes, so we needed to transpile them.

Now its all supported by Vite, so we can drop those plugins.

renchap commented on config/initializers/content_security_policy.rb:
```diff
@@ -32,16 +31,18 @@
   if Rails.env.development?
-    webpacker_public_host = ENV.fetch('WEBPACKER_DEV_SERVER_PUBLIC', Webpacker.config.dev_server[:public])
-    front_end_build_urls = %w(ws http).map { |protocol| "#{protocol}#{Webpacker.dev_server.https? ? 's' : ''}://#{webpacker_public_host}" }
+    # Hacky solution to force CSP to correctly allow localhost, even if ViteRuby is bound to 0.0.0.0.
```
Wr probably need to ensure this is done correctly

ChaosExAnima replied:
how would you recommend we do that?

renchap commented on config/vite/plugin-manifest-sri.ts:
```diff
@@ -0,0 +1,85 @@
+import { createHash } from 'node:crypto';
```
Do we need this file still?

renchap commented on app/javascript/mastodon/performance.js:
```diff
@@ -1,7 +1,6 @@
 // Open up Chrome Dev Tools, then Timeline, then User Timing to see output.
-// Also see config/webpack/loaders/mark.js for the webpack loader marks.
 
 import * as marky from 'marky';
```
As we are removing the `mark-loader` that was in the webpack config, does `marky` still does something? Should it be removed?

renchap commented on vite.config.mts:
```diff
@@ -0,0 +1,142 @@
+    },
+    resolve: {
+      alias: {
+        mastodon: path.resolve(jsRoot, 'mastodon'),
```
Do we still need this, or does Vite uses the aliases from `tsconfig.json`?

ChaosExAnima replied:
Vite does not pull aliases from there, no. there are plugins if we want that.

renchap replied:
What do you think? I have a feeling that if we can have everything defined in only one place (`tsconfig`) then it's cleaner and will cause less issues in the future if we add or remove aliases?

ChaosExAnima replied:
i think it's worth doing, but if it takes more time than expected to just manually copy- we don't have a lot of aliases, after all.

renchap commented on vite.config.mts:
```diff
@@ -0,0 +1,142 @@
+              MastodonServiceWorkerLocales(),
+            ],
+          },
+          // Force the output location, because we have a symlink in `public/sw.js`
```
Should be on the next line

renchap commented on vite.config.mts:
```diff
@@ -0,0 +1,142 @@
+      svgr(),
+      // Old library types need to be converted
+      optimizeLodashImports() as PluginOption,
+      !!process.env.ANALYZE_BUNDLE_SIZE && analyzer({ analyzerMode: 'static' }),
```
`rollup-plugin-visualizer` is a better plugin for this from what I tested

renchap reviewed (commented):
We need to test:
- pages with `two_factor_authentication.ts`
- pages with `sign_up.ts`
- emails

ChaosExAnima committed:
Addresses PR feedback

renchap committed:
Add a way to preload any Vite file (#34650)

ChaosExAnima committed:
add missing i18n message

ChaosExAnima committed:
changes emoji_compressed to be a virtual module and removes preval

ChaosExAnima committed:
use TS for definitive aliases

renchap commented on app/javascript/mastodon/actions/notifications.js:
```diff
@@ -21,6 +21,7 @@ export const NOTIFICATIONS_SET_BROWSER_PERMISSION = 'NOTIFICATIONS_SET_BROWSER_P
 
 defineMessages({
   mention: { id: 'notification.mention', defaultMessage: '{name} mentioned you' },
+  mentioned_you: { id: 'notification.mentioned_you', defaultMessage: '{name} mentioned you' },
```
I dont think this solves the issue, it simply defines this key, but it is not used in the service worker

renchap replied:
See #34668

vmstan commented:
This is from a handful of commits back, I'm about to push the latest so ignore me if this was fixed in the last few, but having a user getting 500 errors.

```
puma-9977b9565-qg2fn [60f577d32ad4de5266fff8e9a982c94d]     25: 
puma-9977b9565-qg2fn [60f577d32ad4de5266fff8e9a982c94d]     26:     %title= html_title
puma-9977b9565-qg2fn [60f577d32ad4de5266fff8e9a982c94d]     27: 
puma-9977b9565-qg2fn [60f577d32ad4de5266fff8e9a982c94d]     28:     = theme_style_tags current_theme
puma-9977b9565-qg2fn [60f577d32ad4de5266fff8e9a982c94d]     29:     = vite_client_tag
puma-9977b9565-qg2fn [60f577d32ad4de5266fff8e9a982c94d]     30:     = vite_react_refresh_tag
puma-9977b9565-qg2fn [60f577d32ad4de5266fff8e9a982c94d]     31:     -# Needed for the wicg-inert polyfill. It needs to be on it's own <style> tag, with this `id`
puma-9977b9565-qg2fn [60f577d32ad4de5266fff8e9a982c94d]   
puma-9977b9565-qg2fn [60f577d32ad4de5266fff8e9a982c94d] lib/vite_ruby/sri_extensions.rb:41:in 'ViteRuby::ManifestIntegrityExtension#missing_entry_error'
puma-9977b9565-qg2fn [60f577d32ad4de5266fff8e9a982c94d] lib/vite_ruby/sri_extensions.rb:5:in 'ViteRuby::ManifestIntegrityExtension#path_and_integrity_for'
puma-9977b9565-qg2fn [60f577d32ad4de5266fff8e9a982c94d] lib/vite_ruby/sri_extensions.rb:100:in 'block (2 levels) in ViteRails::TagHelpers::IntegrityExtension#vite_stylesheet_tag'
puma-9977b9565-tcck7 
puma-9977b9565-tcck7 
puma-9977b9565-tcck7 ):
puma-9977b9565-tcck7 [2e17eb56b320486b17fb5ca75eae197d]     25: 
puma-9977b9565-tcck7 [2e17eb56b320486b17fb5ca75eae197d]     26:     %title= html_title
puma-9977b9565-tcck7 [2e17eb56b320486b17fb5ca75eae197d]     27: 
puma-9977b9565-tcck7 [2e17eb56b320486b17fb5ca75eae197d]     28:     = theme_style_tags current_theme
puma-9977b9565-tcck7 [2e17eb56b320486b17fb5ca75eae197d]     29:     = vite_client_tag
puma-9977b9565-tcck7 [2e17eb56b320486b17fb5ca75eae197d]     30:     = vite_react_refresh_tag
puma-9977b9565-tcck7 [2e17eb56b320486b17fb5ca75eae197d]     31:     -# Needed for the wicg-inert polyfill. It needs to be on it's own <style> tag, with this `id`
puma-9977b9565-tcck7 [2e17eb56b320486b17fb5ca75eae197d]   
puma-9977b9565-tcck7 [2e17eb56b320486b17fb5ca75eae197d] lib/vite_ruby/sri_extensions.rb:41:in 'ViteRuby::ManifestIntegrityExtension#missing_entry_error'
puma-9977b9565-tcck7 [2e17eb56b320486b17fb5ca75eae197d] lib/vite_ruby/sri_extensions.rb:5:in 'ViteRuby::ManifestIntegrityExtension#path_and_integrity_for'
puma-9977b9565-qg2fn [60f577d32ad4de5266fff8e9a982c94d] lib/vite_ruby/sri_extensions.rb:99:in 'Array#each'
puma-9977b9565-qg2fn [60f577d32ad4de5266fff8e9a982c94d] lib/vite_ruby/sri_extensions.rb:99:in 'block in ViteRails::TagHelpers::IntegrityExtension#vite_stylesheet_tag'
puma-9977b9565-qg2fn [60f577d32ad4de5266fff8e9a982c94d] lib/vite_ruby/sri_extensions.rb:98:in 'ViteRails::TagHelpers::IntegrityExtension#vite_stylesheet_tag'
puma-9977b9565-qg2fn [60f577d32ad4de5266fff8e9a982c94d] app/helpers/theme_helper.rb:11:in 'ThemeHelper#theme_style_tags'
puma-9977b9565-qg2fn [60f577d32ad4de5266fff8e9a982c94d] app/views/layouts/application.html.haml:28
puma-9977b9565-qg2fn [60f577d32ad4de5266fff8e9a982c94d] app/controllers/concerns/localized.rb:11:in 'Localized#set_locale'
puma-9977b9565-qg2fn [60f577d32ad4de5266fff8e9a982c94d] lib/mastodon/middleware/socket_cleanup.rb:11:in 'Mastodon::Middleware::SocketCleanup#call'
puma-9977b9565-qg2fn [60f577d32ad4de5266fff8e9a982c94d] lib/mastodon/middleware/public_file_server.rb:20:in 'Mastodon::Middleware::PublicFileServer#call'
puma-9977b9565-qg2fn [60f577d32ad4de5266fff8e9a982c94d] 
puma-9977b9565-qg2fn Information for cause: ViteRuby::MissingEntrypointError (Vite Ruby can't find styles/default.scss in the manifests.
puma-9977b9565-qg2fn 
puma-9977b9565-tcck7 [2e17eb56b320486b17fb5ca75eae197d] lib/vite_ruby/sri_extensions.rb:100:in 'block (2 levels) in ViteRails::TagHelpers::IntegrityExtension#vite_stylesheet_tag'
puma-9977b9565-tcck7 [2e17eb56b320486b17fb5ca75eae197d] lib/vite_ruby/sri_extensions.rb:99:in 'Array#each'
puma-9977b9565-qg2fn Possible causes:
puma-9977b9565-qg2fn   - The file path is incorrect.
puma-9977b9565-qg2fn   - The file is not in the entrypoints directory.
puma-9977b9565-qg2fn   - Some files are outside the sourceCodeDir, and have not been added to watchAdditionalPaths.
puma-9977b9565-qg2fn   - You have not run `bin/vite dev` to start Vite, or the dev server is not reachable.
puma-9977b9565-tcck7 [2e17eb56b320486b17fb5ca75eae197d] lib/vite_ruby/sri_extensions.rb:99:in 'block in ViteRails::TagHelpers::IntegrityExtension#vite_stylesheet_tag'
puma-9977b9565-qg2fn   - "autoBuild" is set to `false` in your config/vite.json for this environment.
puma-9977b9565-qg2fn 
puma-9977b9565-qg2fn Visit the Troubleshooting guide for more information:
puma-9977b9565-tcck7 [2e17eb56b320486b17fb5ca75eae197d] lib/vite_ruby/sri_extensions.rb:98:in 'ViteRails::TagHelpers::IntegrityExtension#vite_stylesheet_tag'
puma-9977b9565-qg2fn   https://vite-ruby.netlify.app/guide/troubleshooting.html#troubleshooting
puma-9977b9565-qg2fn 
puma-9977b9565-qg2fn 
puma-9977b9565-qg2fn 
puma-9977b9565-qg2fn 
puma-9977b9565-qg2fn ):
puma-9977b9565-tcck7 [2e17eb56b320486b17fb5ca75eae197d] app/helpers/theme_helper.rb:11:in 'ThemeHelper#theme_style_tags'
puma-9977b9565-tcck7 [2e17eb56b320486b17fb5ca75eae197d] app/views/layouts/application.html.haml:28
puma-9977b9565-tcck7 [2e17eb56b320486b17fb5ca75eae197d] app/controllers/concerns/localized.rb:11:in 'Localized#set_locale'
puma-9977b9565-tcck7 [2e17eb56b320486b17fb5ca75eae197d] lib/mastodon/middleware/socket_cleanup.rb:11:in 'Mastodon::Middleware::SocketCleanup#call'
puma-9977b9565-tcck7 [2e17eb56b320486b17fb5ca75eae197d] lib/mastodon/middleware/public_file_server.rb:20:in 'Mastodon::Middleware::PublicFileServer#call'
puma-9977b9565-tcck7 [2e17eb56b320486b17fb5ca75eae197d] 
puma-9977b9565-tcck7 Information for cause: ViteRuby::MissingEntrypointError (Vite Ruby can't find styles/default.scss in the manifests.
puma-9977b9565-tcck7 
puma-9977b9565-tcck7 Possible causes:
puma-9977b9565-tcck7   - The file path is incorrect.
puma-9977b9565-tcck7   - The file is not in the entrypoints directory.
puma-9977b9565-tcck7   - Some files are outside the sourceCodeDir, and have not been added to watchAdditionalPaths.
puma-9977b9565-tcck7   - You have not run `bin/vite dev` to start Vite, or the dev server is not reachable.
puma-9977b9565-tcck7   - "autoBuild" is set to `false` in your config/vite.json for this environment.
puma-9977b9565-tcck7 
puma-9977b9565-tcck7 Visit the Troubleshooting guide for more information:
puma-9977b9565-tcck7   https://vite-ruby.netlify.app/guide/troubleshooting.html#troubleshooting
puma-9977b9565-tcck7 
puma-9977b9565-tcck7 
puma-9977b9565-tcck7 
puma-9977b9565-tcck7 
puma-9977b9565-tcck7 ):
puma-9977b9565-tcck7 [2e17eb56b320486b17fb5ca75eae197d]   
puma-9977b9565-tcck7 [2e17eb56b320486b17fb5ca75eae197d] lib/vite_ruby/sri_extensions.rb:41:in 'ViteRuby::ManifestIntegrityExtension#missing_entry_error'
puma-9977b9565-qg2fn [60f577d32ad4de5266fff8e9a982c94d]   
puma-9977b9565-qg2fn [60f577d32ad4de5266fff8e9a982c94d] lib/vite_ruby/sri_extensions.rb:41:in 'ViteRuby::ManifestIntegrityExtension#missing_entry_error'
puma-9977b9565-qg2fn [60f577d32ad4de5266fff8e9a982c94d] lib/vite_ruby/sri_extensions.rb:5:in 'ViteRuby::ManifestIntegrityExtension#path_and_integrity_for'
puma-9977b9565-qg2fn [60f577d32ad4de5266fff8e9a982c94d] lib/vite_ruby/sri_extensions.rb:100:in 'block (2 levels) in ViteRails::TagHelpers::IntegrityExtension#vite_stylesheet_tag'
puma-9977b9565-qg2fn [60f577d32ad4de5266fff8e9a982c94d] lib/vite_ruby/sri_extensions.rb:99:in 'Array#each'
puma-9977b9565-qg2fn [60f577d32ad4de5266fff8e9a982c94d] lib/vite_ruby/sri_extensions.rb:99:in 'block in ViteRails::TagHelpers::IntegrityExtension#vite_stylesheet_tag'
puma-9977b9565-qg2fn [60f577d32ad4de5266fff8e9a982c94d] lib/vite_ruby/sri_extensions.rb:98:in 'ViteRails::TagHelpers::IntegrityExtension#vite_stylesheet_tag'
puma-9977b9565-qg2fn [60f577d32ad4de5266fff8e9a982c94d] app/helpers/theme_helper.rb:11:in 'ThemeHelper#theme_style_tags'
puma-9977b9565-qg2fn [60f577d32ad4de5266fff8e9a982c94d] app/views/layouts/application.html.haml:28
puma-9977b9565-qg2fn [60f577d32ad4de5266fff8e9a982c94d] app/controllers/concerns/localized.rb:11:in 'Localized#set_locale'
puma-9977b9565-qg2fn [60f577d32ad4de5266fff8e9a982c94d] lib/mastodon/middleware/socket_cleanup.rb:11:in 'Mastodon::Middleware::SocketCleanup#call'
puma-9977b9565-qg2fn [60f577d32ad4de5266fff8e9a982c94d] lib/mastodon/middleware/public_file_server.rb:20:in 'Mastodon::Middleware::PublicFileServer#call'
puma-9977b9565-tcck7 [2e17eb56b320486b17fb5ca75eae197d] lib/vite_ruby/sri_extensions.rb:5:in 'ViteRuby::ManifestIntegrityExtension#path_and_integrity_for'
puma-9977b9565-tcck7 [2e17eb56b320486b17fb5ca75eae197d] lib/vite_ruby/sri_extensions.rb:100:in 'block (2 levels) in ViteRails::TagHelpers::IntegrityExtension#vite_stylesheet_tag'
puma-9977b9565-tcck7 [2e17eb56b320486b17fb5ca75eae197d] lib/vite_ruby/sri_extensions.rb:99:in 'Array#each'
puma-9977b9565-tcck7 [2e17eb56b320486b17fb5ca75eae197d] lib/vite_ruby/sri_extensions.rb:99:in 'block in ViteRails::TagHelpers::IntegrityExtension#vite_stylesheet_tag'
puma-9977b9565-tcck7 [2e17eb56b320486b17fb5ca75eae197d] lib/vite_ruby/sri_extensions.rb:98:in 'ViteRails::TagHelpers::IntegrityExtension#vite_stylesheet_tag'
puma-9977b9565-tcck7 [2e17eb56b320486b17fb5ca75eae197d] app/helpers/theme_helper.rb:11:in 'ThemeHelper#theme_style_tags'
puma-9977b9565-tcck7 [2e17eb56b320486b17fb5ca75eae197d] app/views/layouts/application.html.haml:28
puma-9977b9565-tcck7 [2e17eb56b320486b17fb5ca75eae197d] app/controllers/concerns/localized.rb:11:in 'Localized#set_locale'
puma-9977b9565-tcck7 [2e17eb56b320486b17fb5ca75eae197d] lib/mastodon/middleware/socket_cleanup.rb:11:in 'Mastodon::Middleware::SocketCleanup#call'
puma-9977b9565-tcck7 [2e17eb56b320486b17fb5ca75eae197d] lib/mastodon/middleware/public_file_server.rb:20:in 'Mastodon::Middleware::PublicFileServer#call'
puma-9977b9565-tcck7 W, [2025-05-09T14:00:08.039686 #37[]  WARN -- : [5875349b6696985b77ba330e37d17821] Error parsing Collection-Synchronization header
```

vmstan commented:
Still triggering on latest commit.

```
puma-6c9969cccb-kjkrr [6] Puma starting in cluster mode...
puma-6c9969cccb-kjkrr [6] * Puma version: 6.6.0 ("Return to Forever")
puma-6c9969cccb-kjkrr [6[] * Ruby version: ruby 3.4.3 (2025-04-14 revision d0b7e5b6a0) +YJIT +PRISM [x86_64-linux]
puma-6c9969cccb-kjkrr [6] *  Min threads: 16
puma-6c9969cccb-kjkrr [6] *  Max threads: 16
puma-6c9969cccb-kjkrr [6] *  Environment: production
puma-6c9969cccb-kjkrr [6] *   Master PID: 6
puma-6c9969cccb-kjkrr [6] *      Workers: 4
puma-6c9969cccb-kjkrr [6] *     Restarts: (✔) hot (✖) phased
puma-6c9969cccb-kjkrr [6] * Preloading application
puma-6c9969cccb-kjkrr [6] * Listening on http://0.0.0.0:3000
puma-6c9969cccb-kjkrr [6] Use Ctrl-C to stop
puma-6c9969cccb-kjkrr [6] - Worker 0 (PID: 18) booted in 0.02s, phase: 0
puma-6c9969cccb-kjkrr [6] - Worker 1 (PID: 22) booted in 0.01s, phase: 0
puma-6c9969cccb-kjkrr [6] - Worker 2 (PID: 43) booted in 0.01s, phase: 0
puma-6c9969cccb-kjkrr [6] - Worker 3 (PID: 74) booted in 0.0s, phase: 0
puma-6c9969cccb-wb2jl [7] Puma starting in cluster mode...
puma-6c9969cccb-wb2jl [7] * Puma version: 6.6.0 ("Return to Forever")
puma-6c9969cccb-wb2jl [7[] * Ruby version: ruby 3.4.3 (2025-04-14 revision d0b7e5b6a0) +YJIT +PRISM [x86_64-linux]
puma-6c9969cccb-wb2jl [7] *  Min threads: 16
puma-6c9969cccb-wb2jl [7] *  Max threads: 16
puma-6c9969cccb-wb2jl [7] *  Environment: production
puma-6c9969cccb-wb2jl [7] *   Master PID: 7
puma-6c9969cccb-wb2jl [7] *      Workers: 4
puma-6c9969cccb-wb2jl [7] *     Restarts: (✔) hot (✖) phased
puma-6c9969cccb-wb2jl [7] * Preloading application
puma-6c9969cccb-wb2jl [7] * Listening on http://0.0.0.0:3000
puma-6c9969cccb-wb2jl [7] Use Ctrl-C to stop
puma-6c9969cccb-wb2jl [7] - Worker 0 (PID: 19) booted in 0.02s, phase: 0
puma-6c9969cccb-wb2jl [7] - Worker 1 (PID: 24) booted in 0.02s, phase: 0
puma-6c9969cccb-wb2jl [7] - Worker 2 (PID: 29) booted in 0.01s, phase: 0
puma-6c9969cccb-wb2jl [7] - Worker 3 (PID: 34) booted in 0.01s, phase: 0
puma-6c9969cccb-kjkrr E, [2025-05-09T14:16:08.711052 #74[] ERROR -- : [01325cae77fe2583377ff3fbf5694e98]   
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98] ActionView::Template::Error (Vite Ruby can't find styles/default.scss in the manifests.
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr Possible causes:
puma-6c9969cccb-kjkrr   - The file path is incorrect.
puma-6c9969cccb-kjkrr   - The file is not in the entrypoints directory.
puma-6c9969cccb-kjkrr   - Some files are outside the sourceCodeDir, and have not been added to watchAdditionalPaths.
puma-6c9969cccb-kjkrr   - You have not run `bin/vite dev` to start Vite, or the dev server is not reachable.
puma-6c9969cccb-kjkrr   - "autoBuild" is set to `false` in your config/vite.json for this environment.
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr Visit the Troubleshooting guide for more information:
puma-6c9969cccb-kjkrr   https://vite-ruby.netlify.app/guide/troubleshooting.html#troubleshooting
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr )
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98] Caused by: ViteRuby::MissingEntrypointError (Vite Ruby can't find styles/default.scss in the manifests.
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr Possible causes:
puma-6c9969cccb-kjkrr   - The file path is incorrect.
puma-6c9969cccb-kjkrr   - The file is not in the entrypoints directory.
puma-6c9969cccb-kjkrr   - Some files are outside the sourceCodeDir, and have not been added to watchAdditionalPaths.
puma-6c9969cccb-kjkrr   - You have not run `bin/vite dev` to start Vite, or the dev server is not reachable.
puma-6c9969cccb-kjkrr   - "autoBuild" is set to `false` in your config/vite.json for this environment.
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr Visit the Troubleshooting guide for more information:
puma-6c9969cccb-kjkrr   https://vite-ruby.netlify.app/guide/troubleshooting.html#troubleshooting
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr )
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98] 
puma-6c9969cccb-kjkrr Information for: ActionView::Template::Error (Vite Ruby can't find styles/default.scss in the manifests.
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr Possible causes:
puma-6c9969cccb-kjkrr   - The file path is incorrect.
puma-6c9969cccb-kjkrr   - The file is not in the entrypoints directory.
puma-6c9969cccb-kjkrr   - Some files are outside the sourceCodeDir, and have not been added to watchAdditionalPaths.
puma-6c9969cccb-kjkrr   - You have not run `bin/vite dev` to start Vite, or the dev server is not reachable.
puma-6c9969cccb-kjkrr   - "autoBuild" is set to `false` in your config/vite.json for this environment.
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr Visit the Troubleshooting guide for more information:
puma-6c9969cccb-kjkrr   https://vite-ruby.netlify.app/guide/troubleshooting.html#troubleshooting
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr ):
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98]     25: 
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98]     26:     %title= html_title
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98]     27: 
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98]     28:     = theme_style_tags current_theme
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98]     29:     = vite_client_tag
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98]     30:     = vite_react_refresh_tag
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98]     31:     -# Needed for the wicg-inert polyfill. It needs to be on it's own <style> tag, with this `id`
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98]   
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98] lib/vite_ruby/sri_extensions.rb:41:in 'ViteRuby::ManifestIntegrityExtension#missing_entry_error'
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98] lib/vite_ruby/sri_extensions.rb:5:in 'ViteRuby::ManifestIntegrityExtension#path_and_integrity_for'
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98] lib/vite_ruby/sri_extensions.rb:100:in 'block (2 levels) in ViteRails::TagHelpers::IntegrityExtension#vite_stylesheet_tag'
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98] lib/vite_ruby/sri_extensions.rb:99:in 'Array#each'
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98] lib/vite_ruby/sri_extensions.rb:99:in 'block in ViteRails::TagHelpers::IntegrityExtension#vite_stylesheet_tag'
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98] lib/vite_ruby/sri_extensions.rb:98:in 'ViteRails::TagHelpers::IntegrityExtension#vite_stylesheet_tag'
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98] app/helpers/theme_helper.rb:11:in 'ThemeHelper#theme_style_tags'
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98] app/views/layouts/application.html.haml:28
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98] app/controllers/concerns/localized.rb:11:in 'Localized#set_locale'
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98] lib/mastodon/middleware/socket_cleanup.rb:11:in 'Mastodon::Middleware::SocketCleanup#call'
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98] lib/mastodon/middleware/public_file_server.rb:20:in 'Mastodon::Middleware::PublicFileServer#call'
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98] 
puma-6c9969cccb-kjkrr Information for cause: ViteRuby::MissingEntrypointError (Vite Ruby can't find styles/default.scss in the manifests.
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr Possible causes:
puma-6c9969cccb-kjkrr   - The file path is incorrect.
puma-6c9969cccb-kjkrr   - The file is not in the entrypoints directory.
puma-6c9969cccb-kjkrr   - Some files are outside the sourceCodeDir, and have not been added to watchAdditionalPaths.
puma-6c9969cccb-kjkrr   - You have not run `bin/vite dev` to start Vite, or the dev server is not reachable.
puma-6c9969cccb-kjkrr   - "autoBuild" is set to `false` in your config/vite.json for this environment.
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr Visit the Troubleshooting guide for more information:
puma-6c9969cccb-kjkrr   https://vite-ruby.netlify.app/guide/troubleshooting.html#troubleshooting
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr ):
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98]   
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98] lib/vite_ruby/sri_extensions.rb:41:in 'ViteRuby::ManifestIntegrityExtension#missing_entry_error'
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98] lib/vite_ruby/sri_extensions.rb:5:in 'ViteRuby::ManifestIntegrityExtension#path_and_integrity_for'
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98] lib/vite_ruby/sri_extensions.rb:100:in 'block (2 levels) in ViteRails::TagHelpers::IntegrityExtension#vite_stylesheet_tag'
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98] lib/vite_ruby/sri_extensions.rb:99:in 'Array#each'
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98] lib/vite_ruby/sri_extensions.rb:99:in 'block in ViteRails::TagHelpers::IntegrityExtension#vite_stylesheet_tag'
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98] lib/vite_ruby/sri_extensions.rb:98:in 'ViteRails::TagHelpers::IntegrityExtension#vite_stylesheet_tag'
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98] app/helpers/theme_helper.rb:11:in 'ThemeHelper#theme_style_tags'
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98] app/views/layouts/application.html.haml:28
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98] app/controllers/concerns/localized.rb:11:in 'Localized#set_locale'
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98] lib/mastodon/middleware/socket_cleanup.rb:11:in 'Mastodon::Middleware::SocketCleanup#call'
puma-6c9969cccb-kjkrr [01325cae77fe2583377ff3fbf5694e98] lib/mastodon/middleware/public_file_server.rb:20:in 'Mastodon::Middleware::PublicFileServer#call'
puma-6c9969cccb-wb2jl E, [2025-05-09T14:16:10.693850 #29[] ERROR -- : [2b020f951731a1669655104acd00327a]   
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a] ActionView::Template::Error (Vite Ruby can't find styles/default.scss in the manifests.
puma-6c9969cccb-wb2jl 
puma-6c9969cccb-wb2jl Possible causes:
puma-6c9969cccb-wb2jl   - The file path is incorrect.
puma-6c9969cccb-wb2jl   - The file is not in the entrypoints directory.
puma-6c9969cccb-wb2jl   - Some files are outside the sourceCodeDir, and have not been added to watchAdditionalPaths.
puma-6c9969cccb-wb2jl   - You have not run `bin/vite dev` to start Vite, or the dev server is not reachable.
puma-6c9969cccb-wb2jl   - "autoBuild" is set to `false` in your config/vite.json for this environment.
puma-6c9969cccb-wb2jl 
puma-6c9969cccb-wb2jl Visit the Troubleshooting guide for more information:
puma-6c9969cccb-wb2jl   https://vite-ruby.netlify.app/guide/troubleshooting.html#troubleshooting
puma-6c9969cccb-wb2jl 
puma-6c9969cccb-wb2jl 
puma-6c9969cccb-wb2jl 
puma-6c9969cccb-wb2jl 
puma-6c9969cccb-wb2jl )
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a] Caused by: ViteRuby::MissingEntrypointError (Vite Ruby can't find styles/default.scss in the manifests.
puma-6c9969cccb-wb2jl 
puma-6c9969cccb-wb2jl Possible causes:
puma-6c9969cccb-wb2jl   - The file path is incorrect.
puma-6c9969cccb-wb2jl   - The file is not in the entrypoints directory.
puma-6c9969cccb-wb2jl   - Some files are outside the sourceCodeDir, and have not been added to watchAdditionalPaths.
puma-6c9969cccb-wb2jl   - You have not run `bin/vite dev` to start Vite, or the dev server is not reachable.
puma-6c9969cccb-wb2jl   - "autoBuild" is set to `false` in your config/vite.json for this environment.
puma-6c9969cccb-wb2jl 
puma-6c9969cccb-wb2jl Visit the Troubleshooting guide for more information:
puma-6c9969cccb-wb2jl   https://vite-ruby.netlify.app/guide/troubleshooting.html#troubleshooting
puma-6c9969cccb-wb2jl 
puma-6c9969cccb-wb2jl 
puma-6c9969cccb-wb2jl 
puma-6c9969cccb-wb2jl 
puma-6c9969cccb-wb2jl )
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a] 
puma-6c9969cccb-wb2jl Information for: ActionView::Template::Error (Vite Ruby can't find styles/default.scss in the manifests.
puma-6c9969cccb-wb2jl 
puma-6c9969cccb-wb2jl Possible causes:
puma-6c9969cccb-wb2jl   - The file path is incorrect.
puma-6c9969cccb-wb2jl   - The file is not in the entrypoints directory.
puma-6c9969cccb-wb2jl   - Some files are outside the sourceCodeDir, and have not been added to watchAdditionalPaths.
puma-6c9969cccb-wb2jl   - You have not run `bin/vite dev` to start Vite, or the dev server is not reachable.
puma-6c9969cccb-wb2jl   - "autoBuild" is set to `false` in your config/vite.json for this environment.
puma-6c9969cccb-wb2jl 
puma-6c9969cccb-wb2jl Visit the Troubleshooting guide for more information:
puma-6c9969cccb-wb2jl   https://vite-ruby.netlify.app/guide/troubleshooting.html#troubleshooting
puma-6c9969cccb-wb2jl 
puma-6c9969cccb-wb2jl 
puma-6c9969cccb-wb2jl 
puma-6c9969cccb-wb2jl 
puma-6c9969cccb-wb2jl ):
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a]     25: 
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a]     26:     %title= html_title
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a]     27: 
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a]     28:     = theme_style_tags current_theme
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a]     29:     = vite_client_tag
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a]     30:     = vite_react_refresh_tag
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a]     31:     -# Needed for the wicg-inert polyfill. It needs to be on it's own <style> tag, with this `id`
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a]   
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a] lib/vite_ruby/sri_extensions.rb:41:in 'ViteRuby::ManifestIntegrityExtension#missing_entry_error'
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a] lib/vite_ruby/sri_extensions.rb:5:in 'ViteRuby::ManifestIntegrityExtension#path_and_integrity_for'
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a] lib/vite_ruby/sri_extensions.rb:100:in 'block (2 levels) in ViteRails::TagHelpers::IntegrityExtension#vite_stylesheet_tag'
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a] lib/vite_ruby/sri_extensions.rb:99:in 'Array#each'
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a] lib/vite_ruby/sri_extensions.rb:99:in 'block in ViteRails::TagHelpers::IntegrityExtension#vite_stylesheet_tag'
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a] lib/vite_ruby/sri_extensions.rb:98:in 'ViteRails::TagHelpers::IntegrityExtension#vite_stylesheet_tag'
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a] app/helpers/theme_helper.rb:11:in 'ThemeHelper#theme_style_tags'
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a] app/views/layouts/application.html.haml:28
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a] app/controllers/concerns/localized.rb:11:in 'Localized#set_locale'
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a] lib/mastodon/middleware/socket_cleanup.rb:11:in 'Mastodon::Middleware::SocketCleanup#call'
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a] lib/mastodon/middleware/public_file_server.rb:20:in 'Mastodon::Middleware::PublicFileServer#call'
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a] 
puma-6c9969cccb-wb2jl Information for cause: ViteRuby::MissingEntrypointError (Vite Ruby can't find styles/default.scss in the manifests.
puma-6c9969cccb-wb2jl 
puma-6c9969cccb-wb2jl Possible causes:
puma-6c9969cccb-wb2jl   - The file path is incorrect.
puma-6c9969cccb-wb2jl   - The file is not in the entrypoints directory.
puma-6c9969cccb-wb2jl   - Some files are outside the sourceCodeDir, and have not been added to watchAdditionalPaths.
puma-6c9969cccb-wb2jl   - You have not run `bin/vite dev` to start Vite, or the dev server is not reachable.
puma-6c9969cccb-wb2jl   - "autoBuild" is set to `false` in your config/vite.json for this environment.
puma-6c9969cccb-wb2jl 
puma-6c9969cccb-wb2jl Visit the Troubleshooting guide for more information:
puma-6c9969cccb-wb2jl   https://vite-ruby.netlify.app/guide/troubleshooting.html#troubleshooting
puma-6c9969cccb-wb2jl 
puma-6c9969cccb-wb2jl 
puma-6c9969cccb-wb2jl 
puma-6c9969cccb-wb2jl 
puma-6c9969cccb-wb2jl ):
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a]   
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a] lib/vite_ruby/sri_extensions.rb:41:in 'ViteRuby::ManifestIntegrityExtension#missing_entry_error'
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a] lib/vite_ruby/sri_extensions.rb:5:in 'ViteRuby::ManifestIntegrityExtension#path_and_integrity_for'
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a] lib/vite_ruby/sri_extensions.rb:100:in 'block (2 levels) in ViteRails::TagHelpers::IntegrityExtension#vite_stylesheet_tag'
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a] lib/vite_ruby/sri_extensions.rb:99:in 'Array#each'
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a] lib/vite_ruby/sri_extensions.rb:99:in 'block in ViteRails::TagHelpers::IntegrityExtension#vite_stylesheet_tag'
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a] lib/vite_ruby/sri_extensions.rb:98:in 'ViteRails::TagHelpers::IntegrityExtension#vite_stylesheet_tag'
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a] app/helpers/theme_helper.rb:11:in 'ThemeHelper#theme_style_tags'
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a] app/views/layouts/application.html.haml:28
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a] app/controllers/concerns/localized.rb:11:in 'Localized#set_locale'
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a] lib/mastodon/middleware/socket_cleanup.rb:11:in 'Mastodon::Middleware::SocketCleanup#call'
puma-6c9969cccb-wb2jl [2b020f951731a1669655104acd00327a] lib/mastodon/middleware/public_file_server.rb:20:in 'Mastodon::Middleware::PublicFileServer#call'
puma-6c9969cccb-kjkrr E, [2025-05-09T14:16:20.492587 #18[] ERROR -- : [f93dbcb44980b077d23fead9adb2f49b]   
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b] ActionView::Template::Error (Vite Ruby can't find styles/default.scss in the manifests.
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr Possible causes:
puma-6c9969cccb-kjkrr   - The file path is incorrect.
puma-6c9969cccb-kjkrr   - The file is not in the entrypoints directory.
puma-6c9969cccb-kjkrr   - Some files are outside the sourceCodeDir, and have not been added to watchAdditionalPaths.
puma-6c9969cccb-kjkrr   - You have not run `bin/vite dev` to start Vite, or the dev server is not reachable.
puma-6c9969cccb-kjkrr   - "autoBuild" is set to `false` in your config/vite.json for this environment.
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr Visit the Troubleshooting guide for more information:
puma-6c9969cccb-kjkrr   https://vite-ruby.netlify.app/guide/troubleshooting.html#troubleshooting
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr )
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b] Caused by: ViteRuby::MissingEntrypointError (Vite Ruby can't find styles/default.scss in the manifests.
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr Possible causes:
puma-6c9969cccb-kjkrr   - The file path is incorrect.
puma-6c9969cccb-kjkrr   - The file is not in the entrypoints directory.
puma-6c9969cccb-kjkrr   - Some files are outside the sourceCodeDir, and have not been added to watchAdditionalPaths.
puma-6c9969cccb-kjkrr   - You have not run `bin/vite dev` to start Vite, or the dev server is not reachable.
puma-6c9969cccb-kjkrr   - "autoBuild" is set to `false` in your config/vite.json for this environment.
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr Visit the Troubleshooting guide for more information:
puma-6c9969cccb-kjkrr   https://vite-ruby.netlify.app/guide/troubleshooting.html#troubleshooting
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr )
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b] 
puma-6c9969cccb-kjkrr Information for: ActionView::Template::Error (Vite Ruby can't find styles/default.scss in the manifests.
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr Possible causes:
puma-6c9969cccb-kjkrr   - The file path is incorrect.
puma-6c9969cccb-kjkrr   - The file is not in the entrypoints directory.
puma-6c9969cccb-kjkrr   - Some files are outside the sourceCodeDir, and have not been added to watchAdditionalPaths.
puma-6c9969cccb-kjkrr   - You have not run `bin/vite dev` to start Vite, or the dev server is not reachable.
puma-6c9969cccb-kjkrr   - "autoBuild" is set to `false` in your config/vite.json for this environment.
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr Visit the Troubleshooting guide for more information:
puma-6c9969cccb-kjkrr   https://vite-ruby.netlify.app/guide/troubleshooting.html#troubleshooting
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr ):
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b]     25: 
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b]     26:     %title= html_title
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b]     27: 
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b]     28:     = theme_style_tags current_theme
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b]     29:     = vite_client_tag
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b]     30:     = vite_react_refresh_tag
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b]     31:     -# Needed for the wicg-inert polyfill. It needs to be on it's own <style> tag, with this `id`
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b]   
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b] lib/vite_ruby/sri_extensions.rb:41:in 'ViteRuby::ManifestIntegrityExtension#missing_entry_error'
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b] lib/vite_ruby/sri_extensions.rb:5:in 'ViteRuby::ManifestIntegrityExtension#path_and_integrity_for'
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b] lib/vite_ruby/sri_extensions.rb:100:in 'block (2 levels) in ViteRails::TagHelpers::IntegrityExtension#vite_stylesheet_tag'
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b] lib/vite_ruby/sri_extensions.rb:99:in 'Array#each'
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b] lib/vite_ruby/sri_extensions.rb:99:in 'block in ViteRails::TagHelpers::IntegrityExtension#vite_stylesheet_tag'
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b] lib/vite_ruby/sri_extensions.rb:98:in 'ViteRails::TagHelpers::IntegrityExtension#vite_stylesheet_tag'
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b] app/helpers/theme_helper.rb:11:in 'ThemeHelper#theme_style_tags'
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b] app/views/layouts/application.html.haml:28
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b] app/controllers/concerns/localized.rb:11:in 'Localized#set_locale'
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b] lib/mastodon/middleware/socket_cleanup.rb:11:in 'Mastodon::Middleware::SocketCleanup#call'
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b] lib/mastodon/middleware/public_file_server.rb:20:in 'Mastodon::Middleware::PublicFileServer#call'
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b] 
puma-6c9969cccb-kjkrr Information for cause: ViteRuby::MissingEntrypointError (Vite Ruby can't find styles/default.scss in the manifests.
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr Possible causes:
puma-6c9969cccb-kjkrr   - The file path is incorrect.
puma-6c9969cccb-kjkrr   - The file is not in the entrypoints directory.
puma-6c9969cccb-kjkrr   - Some files are outside the sourceCodeDir, and have not been added to watchAdditionalPaths.
puma-6c9969cccb-kjkrr   - You have not run `bin/vite dev` to start Vite, or the dev server is not reachable.
puma-6c9969cccb-kjkrr   - "autoBuild" is set to `false` in your config/vite.json for this environment.
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr Visit the Troubleshooting guide for more information:
puma-6c9969cccb-kjkrr   https://vite-ruby.netlify.app/guide/troubleshooting.html#troubleshooting
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr 
puma-6c9969cccb-kjkrr ):
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b]   
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b] lib/vite_ruby/sri_extensions.rb:41:in 'ViteRuby::ManifestIntegrityExtension#missing_entry_error'
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b] lib/vite_ruby/sri_extensions.rb:5:in 'ViteRuby::ManifestIntegrityExtension#path_and_integrity_for'
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b] lib/vite_ruby/sri_extensions.rb:100:in 'block (2 levels) in ViteRails::TagHelpers::IntegrityExtension#vite_stylesheet_tag'
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b] lib/vite_ruby/sri_extensions.rb:99:in 'Array#each'
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b] lib/vite_ruby/sri_extensions.rb:99:in 'block in ViteRails::TagHelpers::IntegrityExtension#vite_stylesheet_tag'
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b] lib/vite_ruby/sri_extensions.rb:98:in 'ViteRails::TagHelpers::IntegrityExtension#vite_stylesheet_tag'
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b] app/helpers/theme_helper.rb:11:in 'ThemeHelper#theme_style_tags'
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b] app/views/layouts/application.html.haml:28
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b] app/controllers/concerns/localized.rb:11:in 'Localized#set_locale'
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b] lib/mastodon/middleware/socket_cleanup.rb:11:in 'Mastodon::Middleware::SocketCleanup#call'
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b] lib/mastodon/middleware/public_file_server.rb:20:in 'Mastodon::Middleware::PublicFileServer#call'
```

ChaosExAnima commented:
```
puma-6c9969cccb-kjkrr [f93dbcb44980b077d23fead9adb2f49b] ActionView::Template::Error (Vite Ruby can't find styles/default.scss in the manifests.
```

looks like an issue with `default.scss` being loaded still. does this issue happen with 3b2bef772cdfe8ecd9fd230618efe877696090d5 or earlier?

vmstan commented:
I don't have anything on it happening before today.

renchap commented:
It most likely comes from this change: https://github.com/mastodon/mastodon/pull/34450/commits/245ead6908763af87881a95bf09e9b4387a90c96#diff-a71fff0091f0832514afec2d26616a3420e1cdbe8cb940cb8090794b4efa2c68L11

I do think that we need to continue to special case `default` here. You can probably reproduce this by changing your theme to use the dark theme (= the default theme)

ChaosExAnima committed:
fix bug that loaded the wrong styles

ChaosExAnima commented:
> It most likely comes from this change: [245ead6#diff-a71fff0091f0832514afec2d26616a3420e1cdbe8cb940cb8090794b4efa2c68L11](https://github.com/mastodon/mastodon/commit/245ead6908763af87881a95bf09e9b4387a90c96#diff-a71fff0091f0832514afec2d26616a3420e1cdbe8cb940cb8090794b4efa2c68L11)
> 
> I do think that we need to continue to special case `default` here. You can probably reproduce this by changing your theme to use the dark theme (= the default theme)

yep, as i expected. i pushed up a fix, that should resolve things.

vmstan commented:
Applied it, will monitor.

renchap committed:
[Vite] Handle special translation in service worker (#34668)

ChaosExAnima committed:
improve solution for dev mode CSP

ChaosExAnima committed:
re-add common entrypoint and loads it on all Rails pages

renchap commented on app/javascript/entrypoints/application.ts:
```diff
@@ -1,9 +1,8 @@
 import main from 'mastodon/main';
 import { loadPolyfills } from 'mastodon/polyfills';
 
-start();
```
If `common` is included in the app layout, do we really need to call it from every other entrypoint?

ClearlyClaire commented on config/vite.json:
```diff
@@ -1,17 +1,20 @@
   "all": {
     "sourceCodeDir": "app/javascript",
-    "additionalEntrypoints": ["~/{icons,images}/**/*", "~/styles/*.scss"],
+    "additionalEntrypoints": ["~/{fonts,icons,images}/**/*", "~/styles/*.scss"],
```
Isn't the static `~/styles/*.scss` a possibly breaking change from the previous parsing of `config/themes.yml`?

ClearlyClaire commented on app/javascript/mastodon/components/status_action_bar.jsx:
```diff
@@ -9,7 +9,7 @@ import ImmutablePropTypes from 'react-immutable-proptypes';
 import { connect } from 'react-redux';
 
-import BookmarkIcon from '@/material-icons/400-24px/bookmark-fill.svg';
+import BookmarkIcon from '@/material-icons/400-24px/bookmark-fill.svg?react';
```
Extremely minor, but this could be pulled out to another PR.

ChaosExAnima replied:
i don't think it's a problem with Webpacker, but it caused a fatal with Vite

ChaosExAnima committed:
fix style issue

ChaosExAnima committed:
Updates Vite config with entrypoint updates

- Changes to use `themes.yml`
- Removes unused JS entrypoints as they come from Vite Ruby
- Forces usage of vite-plugin-ruby ^5.1 instead of 5.0.0

ChaosExAnima committed:
fix bug where manifestSRI was looking at the wrong manifest paths

ClearlyClaire reviewed (approved):
As discussed, I think `vite-plugin-ruby` and `vite-plugin-rails` mess with the config in a pretty non-intuitive way, which makes the overall setup pretty messy. But it is no worse than with webpack.

As the glitch-soc maintainer, I'm pretty worried about the implications to its theming system, but that is of no concern to Mastodon itself.

I think there are no remaining blockers for Mastodon itself.

renchap reviewed (approved):
🚀
